apps/wistia-videos/src/functions/getVideos.ts

```diff
@@ -35,26 +37,33 @@ export const fetchVideos = async (
 ): Promise<WistiaVideo[]> => {
   const mappedProjects = await Promise.all(
     projectIds.map(async (id: string) => {
-      const project = await (
+      const project: WistiaProject = await (
         await fetch(`https://api.wistia.com/v1/projects/${id}.json`, {
           headers: {
             Authorization: `Bearer ${bearerToken}`,
           },
         })
       ).json();
-      let videos = project.medias;
-      if (project.mediaCount > 100) {
-        // start at 2 since the first page is already gotten
-        for (let i = 2; i <= getTotalPages(project.mediaCount); i++) {
-          const additionalProject = await (
-            await fetch(`https://api.wistia.com/v1/projects/${id}.json?page=${i}`, {
-              headers: {
-                Authorization: `Bearer ${bearerToken}`,
-              },
-            })
-          ).json();
-          videos = [...videos, ...additionalProject.medias];
-        }
+      let videos: WistiaVideo[] = project.medias;
+
+      if (project.mediaCount > VIDEOS_PER_PAGE) {
+        // create additional array so we can use promise.all and pattern match with above
+        const additionalProjectMedias: WistiaVideo[] = await Promise.all(
+          Object.keys(new Array(getTotalPages(project.mediaCount)).fill(0)).map(
+            async (page: string) => {
+              const response: Response =
+                await // start at 2nd page since 1st page is already gotten before this
```

Comment:
Could be the formatting that's confusing me but I think this `await` is not needed